disperser/batcher/batchstore/minibatch_store.go

```diff
@@ -0,0 +1,620 @@
+package batchstore
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	commondynamodb "github.com/Layr-Labs/eigenda/common/aws/dynamodb"
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/Layr-Labs/eigenda/disperser/batcher"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/google/uuid"
+)
+
+const (
+	batchStatusIndexName      = "BatchStatusIndex"
+	batchSKPrefix             = "BATCH#"
+	minibatchSKPrefix         = "MINIBATCH#"
+	dispersalRequestSKPrefix  = "DISPERSAL_REQUEST#"
+	dispersalResponseSKPrefix = "DISPERSAL_RESPONSE#"
+)
+
+type MinibatchStore struct {
+	dynamoDBClient *commondynamodb.Client
+	tableName      string
+	logger         logging.Logger
+	ttl            time.Duration
+}
+
+var _ batcher.MinibatchStore = (*MinibatchStore)(nil)
+
+func NewMinibatchStore(dynamoDBClient *commondynamodb.Client, logger logging.Logger, tableName string, ttl time.Duration) *MinibatchStore {
+	logger.Debugf("creating minibatch store with table %s with TTL: %s", tableName, ttl)
+	return &MinibatchStore{
+		dynamoDBClient: dynamoDBClient,
+		logger:         logger.With("component", "MinibatchStore"),
+		tableName:      tableName,
+		ttl:            ttl,
+	}
+}
+
+func GenerateTableSchema(tableName string, readCapacityUnits int64, writeCapacityUnits int64) *dynamodb.CreateTableInput {
+	return &dynamodb.CreateTableInput{
+		AttributeDefinitions: []types.AttributeDefinition{
+			{
+				AttributeName: aws.String("BatchID"),
+				AttributeType: types.ScalarAttributeTypeS,
+			},
+			{
+				AttributeName: aws.String("SK"),
+				AttributeType: types.ScalarAttributeTypeS,
+			},
+			{
+				AttributeName: aws.String("BatchStatus"),
+				AttributeType: types.ScalarAttributeTypeN,
+			},
+			{
+				AttributeName: aws.String("CreatedAt"),
+				AttributeType: types.ScalarAttributeTypeN,
+			},
+		},
+		KeySchema: []types.KeySchemaElement{
+			{
+				AttributeName: aws.String("BatchID"),
+				KeyType:       types.KeyTypeHash,
+			},
+			{
+				AttributeName: aws.String("SK"),
+				KeyType:       types.KeyTypeRange,
+			},
+		},
+		TableName: aws.String(tableName),
+		GlobalSecondaryIndexes: []types.GlobalSecondaryIndex{
+			{
+				IndexName: aws.String(batchStatusIndexName),
+				KeySchema: []types.KeySchemaElement{
+					{
+						AttributeName: aws.String("BatchStatus"),
+						KeyType:       types.KeyTypeHash,
+					},
+					{
+						AttributeName: aws.String("CreatedAt"),
```

Comment:
Would this index records that don't have `CreatedAt` attribute?

Reply:
No. All items indexed by a GSI must have values for the attributes that are part of the GSI's key schema. If we had used `CreatedAt` (as opposed to `RequestedAt`) for DispersalRequest, DispersalRequests still would not have been included in the GSI because DispersalRequest does not define `BatchStatus`